iOS_SDK/OneSignalSDK/UnitTests/InAppMessagingIntegrationTests.m

```diff
@@ -888,6 +888,51 @@ - (void)testIAMClickedLaunchesPrompt {
     XCTAssertTrue(action.promptActions[0].didAppear);
 }
 
+- (void)testIAMClickedLaunchesLocationPrompt {
+    let message = [OSInAppMessageTestHelper testMessageJsonWithTriggerPropertyName:OS_DYNAMIC_TRIGGER_KIND_SESSION_TIME withId:@"test_id1" withOperator:OSTriggerOperatorTypeLessThan withValue:@10.0];
+
+    let registrationResponse = [OSInAppMessageTestHelper testRegistrationJsonWithMessages:@[message]];
+
+    // the trigger should immediately evaluate to true and should
+    // be shown once the SDK is fully initialized.
+    [OneSignalClientOverrider setMockResponseForRequest:NSStringFromClass([OSRequestRegisterUser class]) withResponse:registrationResponse];
+
+    [UnitTestCommonMethods initOneSignalAndThreadWait];
+
+    NSMutableDictionary *actionJson = [OSInAppMessageTestHelper.testActionJson mutableCopy];
+    [actionJson setValue:@[@"location"] forKey:@"prompts"];
```

Comment:
nit, `actionJson[@"prompts"] = @"location"`  is easier to read and more consistent with other dictionary setting we have